modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java

```diff
@@ -450,4 +449,41 @@ private Response doHttpRequest(final int port, final String path, final String m
             throw new RuntimeException("Could not perform request against couchbase HTTP endpoint ", ex);
         }
     }
+
+    /**
+     * In addition to getting a 200, we need to make sure that all services we need are enabled and available on
+     * the bucket.
+     * <p>
+     *  Fixes the issue observed in https://github.com/testcontainers/testcontainers-java/issues/2993
+     */
+    private class AllServicesEnabledPredicate implements Predicate<String> {
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public boolean test(final String rawConfig) {
+            try {
+                Map<String, Object> parsedConfig = MAPPER.readValue(rawConfig, new TypeReference<Map<String, Object>>() {});
```

Comment:
Consider using `readTree`

Reply:
I second this, less casting to Map/List..

Reply:
I'll get it patched up shortly